spec/models/embedded_ansible_worker_spec.rb

```diff
@@ -12,126 +12,215 @@
+        it "sets the provider default credential" do
+          provider.default_organization = 123
+          expect(cred_collection).to receive(:create!).with(
+            :name         => "ManageIQ Default Credential",
+            :kind         => "ssh",
+            :organization => 123
+          ).and_return(cred_resource)
+
+          subject.ensure_credential(provider, api_connection)
+          expect(provider.default_credential).to eq(13)
+        end
+
+        it "doesn't recreate the credential if one is already set" do
+          provider.default_credential = 2
+          expect(cred_collection).not_to receive(:create!)
+
+          subject.ensure_credential(provider, api_connection)
+        end
+
+        it "creates the organization if one doesn't exist" do
+          expect(org_collection).to receive(:create!).with(
+            :name        => "ManageIQ",
+            :description => "ManageIQ Default Organization"
+          ).and_return(org_resource)
+
+          expect(cred_collection).to receive(:create!).with(
+            :name         => "ManageIQ Default Credential",
+            :kind         => "ssh",
+            :organization => 12
+          ).and_return(cred_resource)
+
+          subject.ensure_credential(provider, api_connection)
+          expect(provider.default_credential).to eq(13)
+          expect(provider.default_organization).to eq(12)
+        end
       end
-    end
 
-    describe "#ensure_host" do
-      it "sets the provider default host" do
-        provider.default_inventory = 234
-        expect(host_collection).to receive(:create!).with(
-          :name      => "localhost",
-          :inventory => 234,
-          :variables => "---\nansible_connection: local\n"
-        ).and_return(host_resource)
-
-        subject.ensure_host(provider, api_connection)
-        expect(provider.default_host).to eq(15)
+      describe "#ensure_inventory" do
+        it "sets the provider default inventory" do
+          provider.default_organization = 123
+          expect(inv_collection).to receive(:create!).with(
+            :name         => "ManageIQ Default Inventory",
+            :organization => 123
+          ).and_return(inv_resource)
+
+          subject.ensure_inventory(provider, api_connection)
+          expect(provider.default_inventory).to eq(14)
+        end
+
+        it "doesn't recreate the inventory if one is already set" do
+          provider.default_inventory = 3
+          expect(inv_collection).not_to receive(:create!)
+
+          subject.ensure_inventory(provider, api_connection)
+        end
+
+        it "creates the organization if one doesn't exist" do
+          expect(org_collection).to receive(:create!).with(
+            :name        => "ManageIQ",
+            :description => "ManageIQ Default Organization"
+          ).and_return(org_resource)
+
+          expect(inv_collection).to receive(:create!).with(
+            :name         => "ManageIQ Default Inventory",
```

Comment:
I'm not happy generally about doing matching on a name that could possibly change, I think I may move some of the name out into a constant and maybe even add a comment about this :scream: